src/Integration.Vsix.UnitTests/SonarLintTagger/TextBufferIssueTrackerTests.cs

```diff
@@ -243,6 +248,98 @@ public void WhenNewIssuesAreFound_ListenersAreUpdated()
             actualMarkers[1].Issue.RuleKey.Should().Be("S333");
         }
 
+        [TestMethod]
+        public void WhenNewIssuesAreFound_FilterIsApplied_ListenersAreUpdated()
+        {
+            // Arrange
+            var inputIssues = new[]
+            {
+                new Issue { RuleKey = "S111", StartLine = 1, EndLine = 1 },
+                new Issue { RuleKey = "S222", StartLine = 2, EndLine = 2 }
+            };
+
+            var issuesToReturnFromFilter = new[]
+            {
+                new DaemonIssueAdapter(
+                    new Issue { RuleKey = "xxx", StartLine = 3 }, "text1", "hash1")
+            };
+
+            SetupIssuesFilter(issuesToReturnFromFilter, out var issuesPassedToFilter);
+
+            var errorListSinkMock1 = RegisterNewErrorListSink();
+
+            // Act
+            ((IIssueConsumer)testSubject).Accept(mockedJavascriptDocumentFooJs.Object.FilePath, inputIssues);
+
+            // Assert
+            // Check the expected issues were passed to the filter
+            issuesPassedToFilter.Count.Should().Be(2);
```

Comment:
Maybe setup the mock to the input, instead of It.IsAny? This way the assertion won't be needed.

Reply:
Because the input isn't passed directly to the mock; it's converted to a list of `IFilteredIssue` first.